src/main/java/com/librato/metrics/client/Sanitizer.java

```diff
@@ -18,14 +18,16 @@ public String apply(String name) {
     };
 
     /**
-     * Metric names restrictions are described <a href="http://dev.librato.com/v1/metrics">here</a>.
+     * Metric name restrictions are described <a href="http://dev.librato.com/v1/metrics">here</a>.
      */
     @SuppressWarnings("unused")
-    public static final Sanitizer LAST_PASS = new Sanitizer() {
-        private final Pattern disallowedCharacters = Pattern.compile("([^A-Za-z0-9.:\\-_]|[\\[\\]]|\\s)");
-        private final int lengthLimit = 256;
+    public static final Sanitizer NAME_SANITIZER = new Sanitizer() {
+        private final int lengthLimit = 64;
```

Comment:
metric names can be up to 255 characters, tag names 64.

Should we keep the existing method to validate metric names and sources (which seem to use the same rules) and add tag name and tag value checkers?

Reply:
or maybe make a generic method (or class) that takes a pattern and a maximum length?

Reply:
Given the minor explosion in combinations of tag/metric/source/values, I think that's a good idea. I didn't realize this code was being used by metrics & sources as well as tags